### ISSUE TITLE
Seek Through Waveform For Overdubbing Recording

### ISSUE THREAD

**Issue by mnearents** [syedhali/EZAudio#120]
Would be awesome if after I recorded some audio, I could seek to a point in the waveform to begin recording again (say for music recording to fix a mistake). An example of this is iOS Voice Memos app. I can record, seek back to any point, then overwrite a portion of the previous recording.

I know there is seeking in the EZAudioFile, but we can't initialize an EZRecorder with an EZAudioFile.

I'm wondering if I record an audio file, open it as an EZAudioFile, perform my seek, store the start time (or frame position), record the new file, then append the two files using the saved start time and the new audio's duration. Then re-open the merged audio as an EZAudioFile and repeat the process? That seems a little hacky to me. Any thoughts?

**Comment by lususvir** [comment]
What happened to the audio_file_refactor branch?

**Comment by lususvir** [comment]
I would like to be able to append audio at the beginning of a file, or at a place where I specify. If the user drags a seeker to the middle, s/he should be able to record from the middle and overwrite what was there.

I was told you can use AudioFileWritePackets to specify a starting point when writing to an audio file. See here: http://stackoverflow.com/a/30196698/1391672

If anyone knows how to do this with EZAudio, whether it be by subclassing the EZRecorder, or whatever, please let me know. This is one of my last major roadblocks.

**Comment by alexgizh** [comment]
@lususvir Did you find a solution to this problem? If so, I'd be really grateful if you shared it here.

**Comment by lususvir** [comment]
No. In my app I have been using 2 separate files, one permanent and one temporary. After each time the user records, the temporary file is appended to the original at a specified position. The problem with this is that with more than 10 seconds of audio, the append takes time, and I don't think doing it in a background thread will help because the file isn't ready for playback or recording until that export happens.

**Comment by alexgizh** [comment]
@lususvir would you be kind enough to share the piece of code used for appending?

**Comment by lususvir** [comment]
Sure, all the code I used can be found here:
https://github.com/syedhali/EZAudio/issues/118

**Comment by alexgizh** [comment]
Thanks a lot! Having no other options for such a small amount of time, this will have to do! :)
